packages/react/src/components/ModalWrapper/ModalWrapper.tsx

```diff
@@ -6,16 +6,50 @@
  */
 
 import PropTypes from 'prop-types';
-import React from 'react';
+import React, { ReactNode } from 'react';
 import Modal from '../Modal';
 import Button from '../Button';
-import { ButtonKinds } from '../../prop-types/types';
 import { warning } from '../../internal/warning';
+import { ButtonKinds } from '../Button';
 import { noopFn } from '../../internal/noopFn';
+import { keys, match } from '../../internal/keyboard';
+
+interface ModalWrapperProps {
+  buttonTriggerClassName?: string;
+  buttonTriggerText?: ReactNode;
+  children?: ReactNode;
+  disabled?: boolean;
+  handleOpen?: React.MouseEventHandler<HTMLButtonElement>;
+  handleSubmit?: React.ReactEventHandler<HTMLElement>;
+  id?: string;
+  modalBeforeContent?: boolean;
+  modalHeading?: string;
+  modalLabel?: string;
+  modalText?: string;
+  onKeyDown?: React.KeyboardEventHandler<HTMLDivElement>;
+  passiveModal?: boolean;
+  preventCloseOnClickOutside?: boolean;
+  primaryButtonText?: string;
+  renderTriggerButtonIcon: React.ElementType;
+  secondaryButtonText?: string;
+  selectorPrimaryFocus?: string;
+  shouldCloseAfterSubmit?: boolean;
+  status?: string;
+  triggerButtonIconDescription?: string;
+  triggerButtonKind: (typeof ButtonKinds)[number];
+  withHeader?: boolean;
```

Comment:
Would you mind copying the comments from the proptypes up into the interface here so that things like intellisense provide the additional usage/info/context?

Reply:
The proptype doesn't have comments that's why I didn't add them here.

Reply:
True! It doesn't have prop comments. Might be because the file is kinda old, 7 years. 🤔

Anyways, in this case we can merge without it and leave as it is.